[EMCAL-791] Require CCDB access for EMCAL digitizer

mfasDa opened the pull request:
EMCAL sim params should be handled via the CCDB,
therefore CCDB access is required for EMCAL.

mfasDa committed:
[EMCAL-791] Require CCDB access for EMCAL digitizer

EMCAL sim params should be handled via the CCDB,
therefore CCDB access is required for EMCAL.

alibuild commented:
Error while checking build/O2/fullCI for 6242cca06e141273c317ae4714381c7b59d8ac95 at 2023-10-05 11:50:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/QualityControl-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[ERROR] Function RecoDecay::getMassPDG is deprecated and will be removed soon.
[ERROR] Please use the Mass function in the O2DatabasePDG service instead.
[ERROR] See the example of usage in Tutorials/src/usingPDGService.cxx.
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12023/6242cca06e141273c317ae4714381c7b59d8ac95/build_O2_fullCI/pretty.html).

mfasDa commented:
CI errors unrelated

davidrohr commented:
Well, unrelated, but let's wait for the FullCI to pass before merging, since we change the full system test.

shahor02 commented:
@davidrohr sorry, saw your message after the merging. But anyway, the PR affects EMC digitization, while the fullCI has failed in the ITS-TPC QC for known reasons well after the digitization.

mfasDa commented:
Something is strange here: After discussion with Sandro I have changed the logics of the digitiser workflow to invert the use-ccdb-emc to no-use-ccdb-emc in order to make the CCDB access the default behaviour, and rebased and force-pushed the commit. But seems it was not updated in this commit.

davidrohr commented:
No idea, then please to this in a separate PR

mfasDa commented:
Probably this was pushed to upstream instead of origin, seems I can create branches in this repository. The change in the digitiser was instead pushed to origin (my fork) and therefore was not updated. Superseding PR https://github.com/AliceO2Group/AliceO2/pull/12029

davidrohr commented:
@ktf @TimoWilken  : Can you check why people may create branches upstream?

TimoWilken commented:
Hi @davidrohr, Markus is a code owner and therefore needs "write" access to this repo. This is why he can push branches to it.

davidrohr commented:
Ok, understood, thx for checking.
